### PR TITLE
Tree wide various cleanups

### PR DESCRIPTION
**Pull request by ssahani**

### CODE REVIEW

File: src/core/namespace.c [hunk 1]
```diff
@@ -464,11 +460,12 @@ static int mount_path_compare(const MountEntry *a, const MountEntry *b) {
 }
 
 static int prefix_where_needed(MountEntry *m, size_t n, const char *root_directory) {
-        size_t i;
-
         /* Prefixes all paths in the bind mount table with the root directory if the entry needs that. */
 
-        for (i = 0; i < n; i++) {
+        assert(m);
+        assert(n);
```
Comment (yuwata) [inline src/core/namespace.c]: Hmm? Maybe, `assert(m || n == 0);`?

File: src/core/target.c [hunk 2]
```diff
@@ -55,7 +54,7 @@ static int target_add_default_dependencies(Target *t) {
         /* Imply ordering for requirement dependencies on target units. Note that when the user created a contradicting
          * ordering manually we won't add anything in here to make sure we don't create a loop. */
 
-        for (k = 0; k < ELEMENTSOF(deps); k++) {
+        for (unsigned k = 0; k < ELEMENTSOF(deps); k++) {
```
Comment (yuwata) [inline src/core/target.c]: `size_t` is better.

File: src/basic/sigbus.c [hunk 3]
```diff
@@ -23,12 +23,10 @@ static void* volatile sigbus_queue[SIGBUS_QUEUE_MAX];
 static volatile sig_atomic_t n_sigbus_queue = 0;
 
 static void sigbus_push(void *addr) {
-        unsigned u;
-
         assert(addr);
 
         /* Find a free place, increase the number of entries and leave, if we can */
-        for (u = 0; u < SIGBUS_QUEUE_MAX; u++)
+        for (unsigned u = 0; u < SIGBUS_QUEUE_MAX; u++)
```
Comment (yuwata) [inline src/basic/sigbus.c]: `size_t` is better.

File: src/coredump/coredump.c [hunk 4]
```diff
@@ -210,14 +210,13 @@ static int fix_xattr(int fd, const Context *context) {
...
         int r = 0;
-        unsigned i;
 
         assert(fd >= 0);
 
         /* Attach some metadata to coredumps via extended
          * attributes. Just because we can. */
 
-        for (i = 0; i < _META_MAX; i++) {
+        for (unsigned i = 0; i < _META_MAX; i++) {
```
Comment (yuwata) [inline src/coredump/coredump.c:219]: `size_t`?

File: src/coredump/coredump.c [hunk 5]
```diff
@@ -818,10 +817,10 @@ static int save_context(Context *context, const struct iovec_wrapper *iovw) {
 
         /* The context does not allocate any memory on its own */
 
-        for (n = 0; n < iovw->count; n++) {
+        for (unsigned n = 0; n < iovw->count; n++) {
```
Comment (yuwata) [inline src/coredump/coredump.c]: Please use `size_t`.

File: src/coredump/coredump.c [hunk 6]
```diff
@@ -818,10 +817,10 @@ static int save_context(Context *context, const struct iovec_wrapper *iovw) {
...
                 struct iovec *iovec = iovw->iovec + n;
 
-                for (i = 0; i < ELEMENTSOF(meta_field_names); i++) {
+                for (unsigned i = 0; i < ELEMENTSOF(meta_field_names); i++) {
```
Comment (yuwata) [inline src/coredump/coredump.c]: `size_t`

**Review by yuwata — CHANGES_REQUESTED** [review]
Could you squash changes "tighten variable scope used in loop"?

File: src/basic/virt.c [hunk 7]
```diff
@@ -159,10 +159,9 @@ static int detect_vm_dmi(void) {
                 /* https://wiki.freebsd.org/bhyve */
                 { "BHYVE",               VIRTUALIZATION_BHYVE     },
         };
-        unsigned i;
         int r;
 
-        for (i = 0; i < ELEMENTSOF(dmi_vendors); i++) {
+        for (unsigned i = 0; i < ELEMENTSOF(dmi_vendors); i++) {
```
Comment (yuwata) [inline src/basic/virt.c]: `size_t`

File: src/core/dbus-execute.c [hunk 8]
```diff
@@ -753,7 +751,7 @@ static int property_get_temporary_filesystems(
         if (r < 0)
                 return r;
 
-        for (i = 0; i < c->n_temporary_filesystems; i++) {
+        for (unsigned i = 0; i < c->n_temporary_filesystems; i++) {
```
Comment (yuwata) [inline src/core/dbus-execute.c:754]: `size_t`

File: src/core/device.c [hunk 9]
```diff
@@ -193,11 +193,10 @@ static const struct {
 
 static int device_found_to_string_many(DeviceFound flags, char **ret) {
         _cleanup_free_ char *s = NULL;
-        unsigned i;
 
         assert(ret);
 
-        for (i = 0; i < ELEMENTSOF(device_found_map); i++) {
+        for (unsigned i = 0; i < ELEMENTSOF(device_found_map); i++) {
```
Comment (yuwata) [inline src/core/device.c]: `size_t`

File: src/libsystemd-network/sd-dhcp-server.c [hunk 10]
```diff
@@ -1057,13 +1057,12 @@ int sd_dhcp_server_start(sd_dhcp_server *server) {
...
 
 int sd_dhcp_server_forcerenew(sd_dhcp_server *server) {
-        unsigned i;
         int r = 0;
 
         assert_return(server, -EINVAL);
         assert(server->bound_leases);
 
-        for (i = 0; i < server->pool_size; i++) {
+        for (unsigned i = 0; i < server->pool_size; i++) {
```
Comment (yuwata) [inline src/libsystemd-network/sd-dhcp-server.c]: `uint32_t`?

**Review by yuwata — CHANGES_REQUESTED** [review]

**Comment by ssahani** [comment]
Thanks for reviewing.  I may have missed something. ptal

**Comment by bluca** [comment]
Please use "treewide: tighten variable scope used in loop" and squashfs all those commits into a single one

**Comment by keszybz** [comment]
LGTM. I'll just squash it.

**Comment by ssahani** [comment]
Thanks for the review @keszybz

**Pull request merged**